src/glue/jobs/compare_parquet_datasets.py

```diff
@@ -0,0 +1,473 @@
+import argparse
+import json
+import os
+import logging
+
+import boto3
+import datacompy
+import pandas as pd
+from pyarrow import fs
+import pyarrow.parquet as pq
+import synapseclient
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+INDEX_FIELD_MAP = {
+    "dataset_enrolledparticipants": ["ParticipantIdentifier"],
+    "dataset_fitbitprofiles": ["ParticipantIdentifier", "ModifiedDate"],
+    "dataset_fitbitdevices": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitactivitylogs": ["LogId"],
+    "dataset_fitbitdailydata": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitintradaycombined": ["ParticipantIdentifier", "Type", "DateTime"],
+    "dataset_fitbitrestingheartrates": ["ParticipantIdentifier", "Date"],
+    "dataset_fitbitsleeplogs": ["LogId"],
+    "dataset_healthkitv2characteristics": ["HealthKitCharacteristicKey"],
+    "dataset_healthkitv2samples": ["HealthKitSampleKey"],
+    "dataset_healthkitv2samples_deleted": ["HealthKitSampleKey"],
+    "dataset_healthkitv2heartbeat": ["HealthKitHeartbeatSampleKey"],
+    "dataset_healthkitv2statistics": ["HealthKitStatisticKey"],
+    "dataset_healthkitv2clinicalrecords": ["HealthKitClinicalRecordKey"],
+    "dataset_healthkitv2electrocardiogram": ["HealthKitECGSampleKey"],
+    "dataset_healthkitv2workouts": ["HealthKitWorkoutKey"],
+    "dataset_healthkitv2activitysummaries": ["HealthKitActivitySummaryKey"],
+    "dataset_googlefitsamples": ["GoogleFitSampleKey"],
+    "dataset_symptomlog": ["DataPointKey"],
+}
+
+
+def read_args():
+    parser = argparse.ArgumentParser(
+        description=(
+            "Compare parquet datasets between two namespaced S3 bucket locations"
+        )
+    )
+    parser.add_argument(
+        "--staging-namespace",
+        required=True,
+        type = validate_args,
+        help="The name of the staging namespace to use",
+        default="staging"
+    )
+    parser.add_argument(
+        "--main-namespace",
+        required=True,
+        type = validate_args,
+        help=("The name of the main namespace to use"),
+        default="main"
+    )
+    parser.add_argument(
+        "--parquet-bucket",
+        required=True,
+        type = validate_args,
+        help=("The name of the S3 bucket containing the S3 files to compare"),
+        default="recover-dev-processed_data"
+    )
```

Comment:
This script assumes a lot of things about where the parquet datasets are located in S3. I think it's fine to leave it as it is now (we don't yet have a use case for needing a more general parameterization and it looks like a hassle to refactor), but I do wish that this script would have been able to accept arbitrary S3 locations of two parquet datasets.

Reply:
I added a ticket to track this ^